src/main/java/de/tum/cit/aet/artemis/communication/web/ForwardedMessageResource.java

```diff
@@ -0,0 +1,104 @@
+package de.tum.cit.aet.artemis.communication.web;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.communication.domain.ForwardedMessage;
+import de.tum.cit.aet.artemis.communication.dto.ForwardedMessageDTO;
+import de.tum.cit.aet.artemis.communication.dto.ForwardedMessagesGroupDTO;
+import de.tum.cit.aet.artemis.communication.repository.ForwardedMessageRepository;
+import de.tum.cit.aet.artemis.core.exception.BadRequestAlertException;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastStudent;
+import de.tum.cit.aet.artemis.core.util.TimeLogUtil;
+
+/**
+ * REST controller for managing ForwardedMessages.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping(value = "api/")
+public class ForwardedMessageResource {
+
+    private static final Logger log = LoggerFactory.getLogger(ForwardedMessageResource.class);
+
+    private static final String ENTITY_NAME = "forwardedMessage";
+
+    private final ForwardedMessageRepository forwardedMessageRepository;
+
+    public ForwardedMessageResource(ForwardedMessageRepository forwardedMessageRepository) {
+        this.forwardedMessageRepository = forwardedMessageRepository;
+    }
+
+    /**
+     * POST /forwarded-messages : Create a new forwarded message.
+     *
+     * @param forwardedMessage the forwarded message to create.
+     * @return the ResponseEntity with status 201 (Created) and with the body containing the new forwarded message,
+     *         or with status 400 (Bad Request) if the forwarded message already has an ID.
+     * @throws BadRequestAlertException if the forwarded message already has an ID.
+     */
+    @PostMapping("forwarded-messages")
+    @EnforceAtLeastStudent
+    public ResponseEntity<ForwardedMessageDTO> createForwardedMessage(@RequestBody ForwardedMessage forwardedMessage) throws URISyntaxException {
```

Comment:
please use a DTO here, we want to avoid using entity objects completely in the REST layer in the future